modules/serverpod_auth/serverpod_auth_server/lib/src/business/users.dart

```diff
@@ -125,6 +124,18 @@ class Users {
       await AuthConfig.current.onUserUpdated!(session, userInfo);
     }
 
+    // Notify if any scopes are revoked for user.
+    if (removedScopes.isNotEmpty) {
+      var removedScopesList =
+          removedScopes.map((s) => s.name).whereType<String>().toList();
```

Comment:
Just for my understanding, are there different type of scopes? String and something else?

Reply:
Scopes is an internal class, but we don't support serializing it. Therefore we need to do this dance of converting scopes to strings and strings to scopes in order to serialize it both to the database and to any communication layer.

Reply:
Ah I see, thanks!